pkg/machines/components/libvirtSlate.jsx

```diff
@@ -0,0 +1,76 @@
+/*jshint esversion: 6 */
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from "react";
+import cockpit from 'cockpit';
+import { mouseClick } from "../helpers.es6";
+import { startLibvirt } from "../actions.es6";
+
+const _ = cockpit.gettext;
+
+const LibvirtSlate = ({ libvirtState, dispatch }) => {
+    const goToLink = () => cockpit.jump("/system/services#/libvirtd.service");
+    const start = () => dispatch(startLibvirt());
+
+    let message;
+    let icon;
+    let detail;
+    let action;
+
+    switch (libvirtState.status) {
+        case 'active':
+            message = _("libvirt is active");
+            icon = (<span className="pficon-ok"/>);
+            break;
+        case 'unknown':
+            message = _("Connecting to libvirt");
+            icon = (<div className="spinner spinner-lg"/>);
+            break;
+        default:
+            message = _("libvirt is not active on the system");
+            icon = (<span className="fa fa-exclamation-circle"/>);
+            detail = (
+                <p>
+                    {_("See libvirt service for ")}<a onClick={mouseClick(goToLink)}>{_("more configuration.")}</a>
```

Comment:
It's fine to say libvirt here and, in fact, would help to reinforce that libvirt is a virtualization service (and vice versa) based on context.

Perhaps the code should look something like the following:

```jsx
            detail = (
                <p> {
                    cockpit.format(_("Please refer to $0 for more information."),
                        `<a href="${goToLink}" target="_blank">libvirt documentation</a>`)
                } </p>
            );
```

(It might not be quite right -- I didn't test it. It's based on the file pkg/ovirt/components/ConsoleClientResources.jsx which has a the link as an actual link (instead of a JavaScript-only action) and the method in which it is written enables the strings to be translatable more easily.)

Reply:
- Docs on `cockpit.format()` are located @ http://cockpit-project.org/guide/latest/cockpit-util
- The reference lines of `ConsoleClientResources.jsx` are @ https://github.com/cockpit-project/cockpit/blob/master/pkg/ovirt/components/ConsoleClientResources.jsx#L34-L35

Reply:
I get the reason to do it, but I do not think it is a good idea.
That example is much easier because it is just a link (outside of cockpit).
- passing function like that breaks things:
`Refused to execute inline event handler because it violates the following Content Security Policy directive: "default-src 'self' http://localhost:9090". Either the 'unsafe-inline' keyword, a hash ('sha256-...'), or a nonce ('nonce-...') is required to enable inline execution.`
- it is not a react like
- translation is problematic; the link can be at various position in other languages

Reply:
I know; this way it is also not translation friendly

Reply:
Yeah, I don't know React*. Please listen to someone else on the team with React experience. Treat any React code I write anytime soon to basically be pseudo-code. :wink: